keras_cv/models/object_detection_3d/center_pillar_backbone.py

```diff
@@ -182,24 +155,58 @@ def UNet(
         down_block_configs: a list of (filter_count, num_blocks) tuples
             indicating the number of filters and sub-blocks in each down block
         up_block_configs: a list of filter counts, one for each up block
-        down_block: a downsampling block
-        up_block: an upsampling block
+        down_block: a Python function with two arguments which returns a
+            curried function to create a downsampling block. See the default
+            `DownSampleBlock` for an example.
+        up_block: a Python function with one argument which returns a curried
+            function to create an upsampling block. See the default
+            `UpSampleBlock` for an example.
     """
 
-    input = layers.Input(shape=input_shape)
-    x = input
-
-    skip_connections = []
-    # Filters refers to the number of convolutional filters in each block,
-    # while num_blocks refers to the number of sub-blocks within a block
-    # (Note that only the first sub-block will perform downsampling)
-    for filters, num_blocks in down_block_configs:
-        skip_connections.append(x)
-        x = down_block(filters, num_blocks, sync_bn=sync_bn)(x)
-
-    for filters in up_block_configs:
-        x = up_block(filters, sync_bn=sync_bn)(x, skip_connections.pop())
-
-    output = x
-
-    return keras.Model(input, output)
+    def __init__(
+        self,
+        down_block_configs,
+        up_block_configs,
+        down_block=DownSampleBlock,
```

Comment:
@fchollet prefers string defaults to classes here.

Reply:
Just removed these entirely, as I agree with your comment above that this is unnecessarily modular.